Source/cryptalgo/Hash.h

```diff
@@ -96,11 +96,27 @@ namespace Crypto {
         SHA1(const SHA1&);
         SHA1& operator=(const SHA1&);
 
+    public:
+        typedef struct {
+            uint32_t H[5]; // Message digest buffers
+
+            uint32_t lengthLow; // Message length in bits
+            uint32_t lengthHigh; // Message length in bits
+
+            uint8_t messageBlock[64]; // 512-bit message blocks
+            uint32_t messageIndex; // Index into message block array
+        } Context;
+
     public:
         inline SHA1()
         {
             Reset();
         }
+        inline SHA1(const uint8_t context[])
```

Comment:
To avoid confusion, why not use the Context as the input parameter. This way the ContextLength() method becomes obsolete and the signature is much clearer, it expects a inline SHA1(const Context& context); and it is used to create a SHA1, from an "existing" context.
Update: As I continued the review there is also a load. Lets drop this method (constructor) and only add the Load(const Context& context); method

Reply:
done